pandas/core/generic.py

```diff
@@ -7148,12 +7148,18 @@ def _clip_with_scalar(self, lower, upper, inplace=False):
             raise ValueError("Cannot use an NA value as a clip threshold")
 
         result = self
-        if upper is not None:
-            subset = self.le(upper, axis=None) | isna(result)
-            result = result.where(subset, upper, axis=None, inplace=False)
-        if lower is not None:
-            subset = self.ge(lower, axis=None) | isna(result)
-            result = result.where(subset, lower, axis=None, inplace=False)
+        mask = isna(self.values)
+
+        with np.errstate(all='ignore'):
+            if upper is not None:
+                subset = self.values <= upper
```

Comment:
Idiomatic approach here would now be `to_numpy` instead of `.values`

Reply:
Updated